projects/igniteui-angular/src/lib/grids/grid/row-drag.directive.spec.ts

```diff
@@ -899,6 +900,46 @@ describe('IgxGrid - Row Drag Tests', () => {
             verifyRowDragEndEvent(dragGrid, rowToDrag, rowDragDirective, false, 3);
         }));
     });
+    describe('Tree Grid Deleting Tests', () => {
+        let dragGrid: IgxTreeGridComponent;
+        let dragRows: DebugElement[];
+        beforeEach(async(() => {
+            fixture = TestBed.createComponent(IgxTreeGridDeleteRowTestComponent);
+            fixture.detectChanges();
+            dragGrid = fixture.componentInstance.treeGrid;
+            dropAreaElement = fixture.debugElement.query(By.directive(IgxDropDirective)).nativeElement;
+            dragIndicatorElements = fixture.debugElement.queryAll(By.css('.' + CSS_CLASS_DRAG_INDICATOR));
+            dragRows = fixture.debugElement.queryAll(By.directive(IgxRowDragDirective));
+        }));
+        configureTestSuite();
+        it('should be able to delete the last root', (async () => {
+            const movePoint: Point = UIInteractions.getPointFromElement(dragGrid.getRowByIndex(2).nativeElement);
+            const dropPoint: Point = UIInteractions.getPointFromElement(dropAreaElement);
+            let error = '';
+
+            spyOn(dragGrid.onRowDragStart, 'emit').and.callThrough();
+            spyOn(dragGrid.onRowDragEnd, 'emit').and.callThrough();
+
+            // delete the last parent node
+            const dragIndicatorElement = dragIndicatorElements[3].nativeElement;
+            const startPoint = UIInteractions.getPointFromElement(dragIndicatorElement);
+
+            await pointerDown(dragIndicatorElement, startPoint, fixture);
+            await pointerMove(dragIndicatorElement, movePoint, fixture);
+            try {
```

Comment:
Wasn't the error caused by having an animation on a row that's being deleted?
Shouldn't the tests wait for that? Otherwise this test might pass even without the fix.

Reply:
Edit: It does ⚠️